extensions/src/main/java/org/pentaho/platform/osgi/KarafBoot.java

```diff
@@ -322,7 +328,13 @@ public static void deleteRecursiveIfExists( File item ) {
         deleteRecursiveIfExists( subitem );
       }
     }
-    item.delete();
+    try {
+      if ( !item.delete() ) {
+        LoggerFactory.getLogger( KarafBoot.class ).warn( item.toURI().toString() + " could not be deleted." );
+      }
+    } catch ( SecurityException exception ) {
+      LoggerFactory.getLogger( KarafBoot.class ).warn( item.toURI().toString() + " cannot delete file. Access denied." );
```

Comment:
Same thing here. Catching an exception and logging the error is not handling the exception. What should happen if we can't delete the file? Can we safely continue executing?